tools/djxl_ng_main.cc

```diff
@@ -621,50 +635,73 @@ bool DecompressJxlToPackedPixelFile(
     }
   }
   boxes.FinalizeOutput();
+  if (decoded_bytes) {
+    *decoded_bytes = compressed.size() - JxlDecoderReleaseInput(dec);
+  }
+  const double t1 = jxl::Now();
+  if (stats) {
+    stats->NotifyElapsed(t1 - t0);
+    stats->SetImageSize(ppf->info.xsize, ppf->info.ysize);
+  }
   return true;
 }
 
 int main(int argc, char** argv) {
-  std::cerr << "Warning: This is work in progress, consider using djxl "
-               "instead!\n";
-
-  gflags::SetUsageMessage("JPEG XL decoder");
-  uint32_t version = JxlDecoderVersion();
-  gflags::SetVersionString(std::to_string(version / 1000000) + "." +
-                           std::to_string((version / 1000) % 1000) + "." +
-                           std::to_string(version % 1000));
-  // TODO(firsching): rethink --help handling
+  std::string version = jpegxl::tools::CodecConfigString(JxlDecoderVersion());
+  gflags::SetVersionString(version);
+  gflags::SetUsageMessage(
+      "JPEG XL decoder\n\n"
+      "Usage: djxl_ng INPUT [OUTPUT] [FLAGS]");
   gflags::ParseCommandLineNonHelpFlags(&argc, &argv, /*remove_flags=*/true);
+  // Disable full help listing, only show flags from this file.
```

Comment:
Potential problems with that approach:

 - the `-help` flag and the `-version` flag are not shown, when running `-help`.
 - personally, I like it when both `--help` and `-h` work for help, perhaps this can be done somehow.
 - (Other than `-h`, the three abbreviations we are dropping are `-s`, `-q` and `-j` and `-V`, which is fine, I guess?!)

Reply:
To get the exact same command-line interface as with djxl, we would need to keep using our own flag parser, since gflags does not have aliases for flags:

From the gflags github page:
"Note that flags do not have single-letter synonyms, like they do in the getopt library, nor do we allow "combining" flags behind a single dash, as in ls -la."

I would be happy to replace the gflags dependency with CommandLineParser.

Reply:
Now I use the same command line parser as djxl, we have -h, -s, -q, -j and -V